src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -87,11 +77,55 @@ public struct MetricPoint
 
         public double DoubleValue { get; internal set; }
 
-        public long[] BucketCounts { get; internal set; }
+        internal MetricPointStatus MetricPointStatus { get; private set; }
 
-        public double[] ExplicitBounds { get; internal set; }
+        public long[] GetBucketCounts()
+        {
+            if (this.aggType == AggregationType.Histogram)
+            {
+                return this.histogramMeasurements.AggregatedBucketCounts;
+            }
+            else
+            {
+                throw new NotSupportedException($"{nameof(this.GetBucketCounts)} is not supported for this metric type.");
+            }
+        }
 
-        internal MetricPointStatus MetricPointStatus { get; private set; }
+        public double[] GetExplicitBounds()
+        {
+            if (this.aggType == AggregationType.Histogram)
+            {
+                return this.histogramMeasurements.ExplicitBounds;
+            }
+            else
+            {
+                throw new NotSupportedException($"{nameof(this.GetExplicitBounds)} is not supported for this metric type.");
+            }
+        }
+
+        public long GetHistogramCount()
+        {
+            if (this.aggType == AggregationType.Histogram || this.aggType == AggregationType.HistogramSumCount)
+            {
+                return this.histogramMeasurements.Count;
```

Comment:
we can simply use the long and double values from MetricPoint itself. No need to add them to HistogramMeasurements.

Reply:
@CodeBlanch was planning to see if we can custom layout to overlap the long/double field. That'd mean we'd be able to still need one of them inside `HistogramMeasurement`.

Reply:
can be addressed as a follow up.